src/main/java/hudson/remoting/JarLoaderImpl.java

```diff
@@ -18,10 +18,9 @@
  */
 @edu.umd.cs.findbugs.annotations.SuppressWarnings("SE_BAD_FIELD")
 class JarLoaderImpl implements JarLoader, Serializable {
-    private final ConcurrentMap<Checksum,URL> knownJars = new ConcurrentHashMap<Checksum,URL>();
+    private final ConcurrentMap<Checksum,URL> knownJars = new ConcurrentHashMap<>();
 
-    @edu.umd.cs.findbugs.annotations.SuppressWarnings("DMI_COLLECTION_OF_URLS") // TODO: fix this
-    private final ConcurrentMap<URL,Checksum> checksums = new ConcurrentHashMap<URL,Checksum>();
+    private final ConcurrentMap<String, Checksum> checksums = new ConcurrentHashMap<>();
```

Comment:
JarLoader is being used as a Channel property, which is accessible on both sides. So it may be really serialized over the channel if the `Channel#remoteChannel` implementation passes it back as a seruializable result